cuml/forest_inference_demo.ipynb

```diff
@@ -0,0 +1,341 @@
+    "        features, labels = make_classification(n_samples=m,\n",
+    "                                               n_features=n,\n",
+    "                                               n_informative=int(n/5),\n",
+    "                                               n_classes=k,\n",
+    "                                               random_state=random_state)\n",
+    "    else:\n",
+    "        features, labels = make_regression(n_samples=m,\n",
+    "                                           n_features=n,\n",
+    "                                           n_informative=int(n/5),\n",
+    "                                           n_targets=1,\n",
+    "                                           random_state=random_state)\n",
+    "    return np.c_[features].astype(np.float32), \\\n",
+    "        np.c_[labels].astype(np.float32).flatten()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### for additional information on the xgboost library please refer to the documentation on : \n",
+    "#### https://xgboost.readthedocs.io/en/latest/parameter.html"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# function that trains the xgboost model and performs prediction on it as well\n",
+    "def train_xgboost_model(X_train, y_train,\n",
+    "                        X_validation,\n",
+    "                        y_validation,\n",
+    "                        num_rounds, classification):\n",
+    "    print(\"Training the xgboost model and saving it to be used for inference\")\n",
+    "    # set the xgboost model parameters\n",
+    "    xgboost_params={}\n",
+    "    params = {'silent': 1}\n",
+    "    if classification:\n",
+    "        params['eval_metric'] = 'error'\n",
+    "        params['objective'] = 'binary:logistic'\n",
+    "    else:\n",
+    "        params['eval_metric'] = 'error'\n",
+    "        params['objective'] = 'reg:squarederror'\n",
+    "        params['base_score'] = 0.0\n",
+    "    params['max_depth'] = 25\n",
+    "    params.update(xgboost_params)\n",
+    "    model_path = \"xgb.model\"\n",
+    "    dtrain = xgb.DMatrix(X_train, label=y_train)\n",
+    "    bst = xgb.train(params, dtrain, num_rounds)\n",
+    "\n",
+    "    # save the trained xgboost model\n",
+    "    bst.save_model(model_path)\n",
+    "\n",
+    "    print(\"Perform validation using the xgboost model and\"\n",
+    "          \" save the predicted output to compare with the Forest Inference library\")\n",
+    "    # predict the xgboost model\n",
+    "    dvalidation = xgb.DMatrix(X_validation, label=y_validation)\n",
+    "    xgb_preds = bst.predict(dvalidation)\n",
+    "\n",
+    "    # if the model is used for classification then convert\n",
+    "    # the predicted values into class labels\n",
+    "    if classification:\n",
+    "        xgb_preds = np.around(xgb_preds)\n",
```

Comment:
It's kind of weird that train_xgboost_model also does prediction... you can either break out prediction to another function or rename to "fit_and_predict_xgboost"

Would be nice to take model_path as a param so it's more emphasized